src/github_runner_image_builder/logging.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Configure logging for GitHub runner image builder."""
+
+import logging
+import logging.handlers
+import pathlib
+
+LOG_FILE_DIR = pathlib.Path.home() / "github-runner-image-builder/log"
+LOG_FILE_PATH = LOG_FILE_DIR / "info.log"
+ERROR_LOG_FILE_PATH = LOG_FILE_DIR / "error.log"
+
+
+def configure(log_level: str | int) -> None:
+    """Configure the global log configurations.
+
+    Args:
+        log_level: The logging verbosity level to apply.
+    """
+    LOG_FILE_DIR.mkdir(parents=True, exist_ok=True)
+    log_handler = logging.handlers.WatchedFileHandler(filename=LOG_FILE_PATH, encoding="utf-8")
```

Comment:
I miss some context, why is the file watched?

As you then are using a chroot environment, if the file "changes" when in the chroot environment, logs inside that  environment could open a file in the chroot environment

Reply:
The logs are not used within the chroot environment - the logs are output in the charm VM even though the commands are running within chroot :)
The same is applied to external VM builder - logs are output to the charm :D